qualtran/bloqs/bookkeeping/auto_partition.py

```diff
@@ -0,0 +1,116 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from functools import cached_property
+from itertools import chain
+from typing import Dict, Sequence, Set, Tuple
+
+from attrs import evolve, field, frozen
+
+from qualtran import (
+    Bloq,
+    bloq_example,
+    BloqBuilder,
+    BloqDocSpec,
+    QAny,
+    Register,
+    Side,
+    Signature,
+    SoquetT,
+)
+from qualtran.bloqs.bookkeeping.partition import Partition
+from qualtran.cirq_interop.t_complexity_protocol import TComplexity
+from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+
+
+@frozen
+class AutoPartition(Bloq):
+    """Wrap a bloq with `Partition` to fit an alternative set of input and output registers
+       such that splits / joins can be avoided in diagrams.
+
+    Args:
+        bloq: The bloq to wrap.
+        partition: A sequence of pairs specifying each register that the wrapped bloq should accept
```

Comment:
should this be a plural name like `paritions` or `partitionings`? If I'm understanding it correctly, each entry says how a given "outer" register gets mapped to the multiple "inner" registers, right?

Reply:
Yes, I will change to `partitions`.